### ISSUE TITLE
Add verbosity levels to logging

### ISSUE THREAD

**Issue by anishnaik** [crytic/medusa#204]
Currently, the execution traces on failed tests can be quite aggressive for more complex systems where we enter into multiple call frames within a single call. Thus, we should introduce various verbosity levels that can allow the user to choose what they want the log output to look like. 

I propose the following log levels:
1. -v: Test results are shown. No execution traces or console.log events are shown
2. -vv: **Top-level** call frame execution traces as well as console.log events are shown for failing tests
3. -vvv: TBD
4. -vvvv: TBD

edit (Jan 31, 2023): There should also be the capability to disable tracing all together (#149 )

**Comment by rappie** [comment]
I would love to see this implemented. My current project has a trace of more than 5000 (!) lines, it's a huge pain to keep scrolling up while debugging :sweat_smile:

**Comment by aviggiano** [comment]
@rappie do you have any workarounds for this? any smart `grep` command?

**Comment by anishnaik** [comment]
We actually will be merging this feature in really really soon :) Currently, no workaround.

**Comment by rappie** [comment]
> [@rappie](https://github.com/rappie) do you have any workarounds for this? any smart `grep` command?

No I don't 🙁 My main workaround is to use Echidna as much as possible for debugging. Some kind of grep command would be awesome 🙏

**Comment by anishnaik** [comment]
Why is echidna better at debugging than medusa for this use case? @rappie

**Comment by rappie** [comment]
> Why is echidna better at debugging than medusa for this use case? [@rappie](https://github.com/rappie)

In general: I'm a big fan of the Echidna UI. It helps give me a better feel for what the fuzzer is doing behind the scenes, for example while shrinking.

For this specific logging issue: The Echidna UI shows the sequence above the trace and the panels are scolled to the top by default, so you see them instantly instead of having to scroll upwards.

**Comment by anishnaik** [comment]
Outside of the TUI (which hopefully we will implement soon), you are saying that the terminal sort of shows the sequence first (before the trace) which makes it easier to reason through what went wrong?

**Comment by rappie** [comment]
Yay for the TUI 😄 

Both the sequence and the trace have their own uses for me.

I generally look at the sequence first to try to understand what the fuzzer is doing. This is enough for 90% of the cases, such as simple tests or obvious false positives.

For deeper debugging I rely on the trace by adding debug prints to the code (simplifying here).

So it's basically a super practical thing: sometimes you want to look at the top of the entire log/terminal/panel to see the trace. The TUI shows this by default. Other times you want to see (near) the bottom, in that case i can exit out of echidna and i'm right there in the terminal log.

**Comment by anishnaik** [comment]
Gotcha, that makes sense. Thank you for explaining :) I guess verbosity levels will get you 60% of the way there bc you can lower the verbosity but if you want all the verbosity then you are stuck with the original problem.

**Comment by rappie** [comment]
Yes. Ideally there would be some way to easily toggle between showing them in the TUI, but I'd already be extremely happy with a simple TUI for now 😆 

Now that you got me started: Another great TUI feature would be to have a filter in the trace so you can show only the emitted events 😇

**Comment by anishnaik** [comment]
Closed by #601
